integrations/access/accessrequest/app.go

```diff
@@ -524,14 +525,22 @@ func (a *App) updateMessages(ctx context.Context, reqID string, tag pd.Resolutio
 func (a *App) getLoginsByRole(ctx context.Context, req types.AccessRequest) (map[string][]string, error) {
 	loginsByRole := make(map[string][]string, len(req.GetRoles()))
 
+	user, err := a.apiClient.GetUser(ctx, req.GetUser(), false)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
 	for _, role := range req.GetRoles() {
 		currentRole, err := a.apiClient.GetRole(ctx, role)
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
-		loginsByRole[role] = currentRole.GetLogins(types.Allow)
+		currentRole, err = services.ApplyTraits(currentRole, user.GetTraits())
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		logins := currentRole.GetLogins(types.Allow)
```

Comment:
Should we care about removing the denied roles?

Reply:
We only grab the allow roles currently, I can't think of a reason to change that